index.bs

```diff
@@ -1507,6 +1506,81 @@ these steps:
 
 </div>
 
+
+## Drag and Drop ## {#drag-and-drop}
+
+<xmp class=idl>
+partial interface DataTransferItem {
+    FileSystemHandle? getAsFileSystemHandle();
+};
+</xmp>
+
+During a <em>drag-and-drop operation</em>, dragged file and
+directory items are associated with [=file entries=] and [=directory entries=] 
+respectively.
+
+<div class="note dominto">
+  : |handle| = item . {{getAsFileSystemEntry()}}
+  :: Returns a {{FileSystemFileHandle}} object if the dragged item is file and a {{FileSystemDirectoryHandle}} object if the dragged item is a directory.
+</div>
+
+<div algorithm>
+
+The <dfn method for=DataTransferItem>getAsFileSystemEntry()</dfn> method steps are:
+
+1. If the {{DataTransferItem}} object is not in the <a spec=html>read/write
+    mode</a> or the <a spec=html>read-only mode</a>, then return null.
+
+1. If the <a spec=html>the drag data item kind</a> is not <em>File</em>, then return
+    null.
+
+1. Let |entry| be the [=entry=] representing the dragged file or directory.
+
+1. Set |entry|'s [=query permission steps=] to the [=native file system query permission steps=]
+
+1. Set |entry|'s [=request permission steps=] to the [=native file system request permission steps=]
+
+1. If |entry| is a [=file entry=]:
+  
+  1. Let |handle| be a {{FileSystemFileHandle}} associated with entry
+
+1. else if |entry| is a [=directory entry=]:
+
+  1. Let |handle| be a {{FileSystemDirectoryHandle}} associated with entry
+
+1. Return |entry|
+
+</div>
+
+<div class=example id=draganddrop-example>
+Handling drag and drop of files and directories:
+<xmp highlight=js>
+elem.addEventListener('dragover', e => {
+  // Prevent navigation.
+  e.preventDefault();
+});
+elem.addEventListener('drop', e => {
+  // Prevent navigation.
+  e.preventDefault();
+
+  // Process all of the items.
+  for (const item of e.dataTransfer.items) {
+    // kind will be 'file' for file/directory entries.
+    if (item.kind === 'file') {
+      const entry = item.getAsFileSystemEntry();
+      if (entry.isDirectory) {
```

Comment:
in the mean time isDirectory/isFile was changed to entry.kind returning 'file' or 'directory'